### PR TITLE
Bridge: change Eth monitoring to manual pull

### PR DESCRIPTION
**Pull request by musitdev**
# Summary
- **RFCs**: [Link to RFC](#./link/to/rfc), [Link to RFC](#./link/to/rfc), or $\emptyset$.
- **Categories**: any of `protocol-units`, `networks`, `scripts`, `util`, `cicd`, or `misc`.

Change the Eth pulling to pull manual using RPC.
# Changelog

# Testing

# Outstanding issues

### CODE REVIEW

File: protocol-units/bridge/integration-tests/tests/client_eth_movement.rs [hunk 1]
```diff
@@ -456,3 +459,103 @@ async fn test_eth_client_should_successfully_complete_transfer() {
 
 	//TODO: Here call complete with the id captured from the event
 }
+
+#[tokio::test]
+async fn test_eth_client_lock_then_complete_transfer() -> Result<(), anyhow::Error> {
```
Comment (andygolay) [inline protocol-units/bridge/integration-tests/tests/client_eth_movement.rs:464]: This test is failing for me against a local Suzuka node. What test do you recommend for confirming that the new Eth monitoring works @musitdev ? The E2E Eth -> Movement test against the framework?

File: protocol-units/bridge/service/src/chains/ethereum/event_monitoring.rs [hunk 2]
```diff
@@ -40,151 +41,241 @@ impl EthMonitoring {
...
+								}
+							}
+							Err(err) => {
+								if sender
+									.send(Err(BridgeContractError::OnChainError(err.to_string())))
+									.await
+									.is_err()
+								{
+									tracing::error!("Failed to send event to listener channel");
+									break;
```
Comment (andygolay) [inline protocol-units/bridge/service/src/chains/ethereum/event_monitoring.rs:132]: After pasting this config in to `bridge/config.yaml`:

```
{
  "eth": {
    "eth_rpc_connection_protocol": "https",
    "eth_rpc_connection_hostname": "ethereum-holesky-rpc.publicnode.com",
    "eth_rpc_connection_port": 443,
    "eth_ws_connection_protocol": "ws",
    "eth_ws_connection_hostname": "0.0.0.0",
    "eth_ws_connection_port": 8090,
    "eth_chain_id": 0,
    "eth_initiator_contract": "0xB7cBE82b07fdA3517AE882e12C7Faa764C16Afb9",
    "eth_counterparty_contract": "0xB5bBBd4913da0a8eb723E8cD1e02d69565f4da66",
    "eth_weth_contract": "0xB7f8BC63BbcaD18155201308C8f3540b07f84F5e",
    "signer_private_key": "0x3128c9024817f3c078695489ff3a389f0e4c27f6d364f663758fa3476564989b",
    "time_lock_secs": 60,
    "gas_limit": 10000000000000000,
    "transaction_send_retries": 10,
    "asset": "MOVE"
  },
  "movement": {
    "movement_signer_key": "0x0000000000000000000000000000000000000000000000000000000000000001",
    "movement_native_address": "0xface",
    "mvt_rpc_connection_protocol": "https",
    "mvt_rpc_connection_hostname": "testnet.bardock.movementnetwork.xyz",
    "mvt_rpc_connection_port": 443,
    "mvt_faucet_connection_protocol": "http",
    "mvt_faucet_connection_hostname": "0.0.0.0",
    "mvt_faucet_connection_port": 30732,
    "mvt_init_network": "custom",
    "rest_hostname": "127.0.0.1",
    "rest_port": 308833,
    "grpc_protocol": "http",
    "grpc_hostname": "127.0.0.1",
    "grpc_port": 50051
  },
  "testing": {
    "eth_well_known_account_private_keys": []
  }
}
```
I get the error
```
[bridge	] 2024-10-25T11:23:01.436574Z  INFO bridge_service: Bridge current transfer processing:[]
[bridge	] 2024-10-25T11:23:15.233736Z ERROR alloy_rpc_client::poller: failed to poll err=server returned an error response: error code -32000: filter not found
[bridge	] 2024-10-25T11:23:15.247208Z ERROR alloy_rpc_client::poller: failed to poll err=server returned an error response: error code -32000: filter not found
[bridge	] 2024-10-25T11:23:22.313824Z ERROR alloy_rpc_client::poller: failed to poll err=server returned an error response: error code -32000: filter not found
[bridge	] 2024-10-25T11:23:22.376266Z ERROR alloy_rpc_client::poller: failed to poll err=server returned an error response: error code -32000: filter not found
[bridge	] 2024-10-25T11:23:22.425767Z ERROR alloy_rpc_client::poller: failed to poll err=server returned an error response: error code -32000: filter not found
[bridge	] thread 'tokio-runtime-worker' panicked at /Users/andygmove/Downloads/repos/movement/protocol-units/bridge/service/src/chains/ethereum/event_monitoring.rs:132:17:
[bridge	] all branches are disabled and there is no else branch
```

where it breaks at this line, 132.

**Comment by 0xmovses** [comment]
The framework e2e test is passing.

**Comment by 0xmovses** [comment]
I cleaned up some commented out code. I can see you've refactored the event monitoring to use the event filters generated from the alloy macro. Looks good let's see how this fares on the live network

**Review by 0xmovses — APPROVED** [review]

File: protocol-units/bridge/service/src/chains/ethereum/event_monitoring.rs (hunk 2 above)
Comment (0xmovses) [reply protocol-units/bridge/service/src/chains/ethereum/event_monitoring.rs:132]: Going to resolve this as I didn't receive this error and the tests passed, not sure what was happening locally for you. I ran the framework e2e test multiple times to be sure.

File: protocol-units/bridge/integration-tests/tests/client_eth_movement.rs (hunk 1 above)
Comment (0xmovses) [reply protocol-units/bridge/integration-tests/tests/client_eth_movement.rs:464]: The e2e framework test is passing for me for framework, I'm keen to get this into production so going to resolve this hope you don't mind

**Pull request merged**